[HOLD] Keep track of notification opt-in status

roryabraham opened the pull request:
HOLD on https://github.com/Expensify/Web-Expensify/pull/34444. Test in parallel with it actually :p
### Details
- Stores an NVP upon login whenever a push notifications are enabled or disabled.
- Does not account for "notification frequency settings" which I _think_ will be added at part of N7 account settings.
- Does not account for users that have different devices. The `pushNotificationsEnabled` NVP is boolean true or false and having different notifications on different mobile devices is not something we're accounting for.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/209233

### Tests (Android & iOS only)
Due to annoying and unpredictable delays in when a push notification is and is not "silent", I recommend changing [this line](https://github.com/Expensify/Web-Expensify/blob/24242ac759e19efc9eb1b2455f19ef2660f4174f/partners/UrbanAirship/push.php#L43) to remove the last boolean param `true`. This will prevent silent notifications and make your testing easier.

1. Tail the logs for `PUSH_NOTIFICATION`
1. Open the mobile app and sign in
1. *iOS only:* tap allow notifications.
1. You should see a log line that says `Push notification opt-in status changed.`
1. Open the dev database and run:

    ```sql
    SELECT *
    FROM nameValuePairs nvp
    INNER JOIN accounts a ON a.accountID = nvp.accountID
    WHERE nvp.name = 'pushNotificationsEnabled'
      AND account .email = '<whatever_email_you_signed_in_with>';
    ```
    You should see that the NVP is `true`.
    
1. *iOS only:* Uninstall the app from the device completely and start again.
1. *iOS only:* Sign in again, and this time _DO NOT_ allow notifications.
1. *iOS only:* Check the database and make sure the NVP is `false` instead of `true`.
1. *iOS only:* Uninstall the app again, reinstall it, sign back in and allow notifications.
1. *iOS only:* Check the database and make sure that the NVP is `true` again.
1. On the mobile device, disable notifications for the New Expensify app in the system settings:
    - *android:* go to `Settings` -> `Apps & Notifications` -> `New Expensify` and disable notifications.
    - *iOS:* `Settings` -> `New Expensify` -> `Notifications` and toggle notifications.
1. Open the app again.
1. Open the dev database and run the same query. You should see that the NVP is now `false`.
1. Re-enable notifications for New Expensify in the device settings.
1. Reopen the app.
1. Open the dev database and run the same query. You should see that the NVP is now `true`.
1. Log out and log in as a different user.
1. Open the dev database and run the same query with the new user's email. You should see that the NVP is now `true`.
1. Back in device settings, disable notifications.
1. Reopen the app.
1. Open the dev database and run the same query with the new user's email. You should see that the NVP is now `false`.
1. Do not reopen the NewDot app.
1. Sign into NewDot web as a different user.
1. Send a message to the mobile user.
1. The mobile user should receive a push notification.
1. Open the dev database and run the same query with the new user's email. You should see that the NVP is now `true`.
1. Put the phone on airplane mode.
1. Disable push notifications in the device settings
1. Open the app.
1. Check the dev database again, the NVP should still be `true`.
1. Turn off airplane mode.
1. Check the dev database again. The NVP should be `false`.

- [x] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.


<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps (Android & iOS only)
### iOS only
1. Sign in to NewDot mobile. When prompted to allow notifications, say _NO_.
1. Sign into OldDot web with the same account.
1. Open the JS console and run `NVP.get('pushNotificationsEnabled');` You should see that the NVP is `false`.
1. Uninstall the app from the iOS device completely and reinstall it.

### iOS and Android
1. Sign in to NewDot mobile.
1. Sign into OldDot web with the same account.
1. Open the JS console and run `NVP.get('pushNotificationsEnabled');`. You should see that the NVP is `true`
1. On the Android device, go to `Settings` -> `Apps & Notifications` -> `New Expensify` and disable notifications.
1. Open the NewDot app again.
1. Back in the OldDot JS console, run the same snippet. You should see that the NVP is now `false`.
1. Re-enable notifications for New Expensify in the device settings.
1. Reopen the NewDot app.
1. Back in the OldDot JS console, run the same snippet. You should see that the NVP is now `true`.
1. Log out and log in as a different user. Do the same on OldDot, so that you're signed into the same new account on NewDot and OldDot.
1. Back in the OldDot JS console, run the same snippet. You should see that the NVP is now `true` for the new user.
1. Back in NewDot device settings, disable notifications.
1. Reopen the app.
1. Back in the OldDot JS console, run the same snippet. You should see that the NVP is now `false`.
1. _DO NOT_ reopen the NewDot app.
1. Sign into _NewDot web_ as a different user.
1. Send a message from the NewDot web user to the NewDot mobile user.
1. The mobile user should receive a push notification.
    - _Note:_ If this doesn't work, please wait 5-10 minutes and try again. Also try locking the Android device. Sorry this is annoying, we have an issue to fix it.
1. Back in the OldDot JS console, run the same snippet. You should see that the NVP is now `true`.
1. Put the phone on airplane mode.
1. Disable push notifications in the device settings
1. Open the NewDot app.
1. Back in the OldDot JS console, run the same snippet. the NVP should still be `true`.
1. Turn off airplane mode.
1. Check the dev database again. The NVP should be `false`.

### IMPORTANT NOTE
@Expensify/applauseleads _**Please create an issue to add these testing steps to the TF regression suite.**_ Thanks!

- [ ] Verify that no errors appear in the JS console

sobitneupane committed:
Fix emoji offset issues

sobitneupane committed:
Fix the height of emoji

arosiclair commented:
Looks pretty good so far. I'm assuming since some of this UA code can't run in the React life cycle, we can't include this as part of the new `OpenApp` & `ReconnectApp` commands?

roryabraham commented:
Just a quick clarification ... I took this out of the README:

> Will "redundantly" save the same NVP value, even if it hasn't changed. I figure this is not a problem for now. If we wanted to, we could add this setting to Onyx and only update the NVP in the server if the new value is different from what's in Onyx.

Because it's not necessarily true anymore (since the NVP is now saved in Onyx)

kadiealexander committed:
Updating copy changes for Request A Call page

kadiealexander committed:
add spanish translations

sobitneupane committed:
Change intitalNumToRender for better performance

sobitneupane committed:
Remove initalNumToRender for better performance

mdneyazahmad committed:
fix: reset grabbed state when mouse is released

MarkCSmith committed:
fix: blue outline appears for a moment on buttons

Remove focus from the chat action button, emoji picker, and the
video chat button immediately after the buttons are clicked/tapped.

aldo-expensify committed:
Wait for policy ids to be reading before sending request

MarkCSmith committed:
fix: blue outline appears for a moment on buttons

Also fix this problem for the FAB.

MarkCSmith committed:
fix: blue outline appears for a moment on buttons

Avoid inline comments.

vladamx committed:
Add splash screen on web and desktop

mdneyazahmad committed:
refactor: rename variable names

Beamanator committed:
Set up new pronouns page route

Beamanator committed:
Add new Pronouns page

Beamanator committed:
Update spanish translation

Beamanator committed:
Fix translation location

Beamanator committed:
Add updatePronouns action

vladamx committed:
Format if statement

Co-authored-by: Rory Abraham <47436092+roryabraham@users.noreply.github.com>

vladamx committed:
Refactor inline media styles for logo sizing

sobitneupane committed:
Change emoji height

tgolen committed:
Add a function to get the contents of the reviewer checklist

tgolen committed:
Add function to get the author checklist items

tgolen committed:
Add specific actions for the author and reviewer checklist

tgolen committed:
Modify workflows for author and reviewer checklists

tgolen committed:
Build out the author checklist action

tgolen committed:
Build out the reviewer checklist action

tgolen committed:
Access author checklist from raw file

tgolen committed:
Remove the old contributor checklist action

tgolen committed:
Add compiled actions

tgolen committed:
Temporarily try to checkout the actions from specific commit hashes

tgolen committed:
Fix regex and add a little more debugging

tgolen committed:
Temporarily update commit hash

tgolen committed:
Update checklists to allow for +/- 2 checkboxes

tgolen committed:
Temporarily bump hashes

tgolen committed:
Fix lint errors

tgolen committed:
Temporarily update hashes

tgolen committed:
Don't check the entire PR body for unfinished checklist items

tgolen committed:
Temporarily bump hashes

tgolen committed:
Move methods to github utils

tgolen committed:
Update references to branch name

tgolen committed:
Update the types of events that trigger this workflow

tgolen committed:
Reference the PR body and build the actions

tgolen committed:
Try to debug why the checklist isn't passing

tgolen committed:
Remove underscore and fix lint

tgolen committed:
Fix a bad variable reference

tgolen committed:
Update actions to checkout code from main

tgolen committed:
Make sure the complete checklist items are case insensitive

tgolen committed:
Add a constant about being careful when changing the name of the job

grgia committed:
1:1 swap to dark mode colors.

youssef-lr committed:
Only update IOU total when network is online

aldo-expensify committed:
Remove incorrect red brick indicator

Puneet-here committed:
refactor to use form

VikalpP committed:
fix: Settings navigation modal leaves empty space at the bottom

luacmartins committed:
create currentDBTime

luacmartins committed:
update functions to use datetime

grgia committed:
Define temp color names in colors.js / use in theme file.

luacmartins committed:
add broken test

luacmartins committed:
more tests

luacmartins committed:
use created

luacmartins committed:
initialize optimistic actions with created date

luacmartins committed:
rm moment import and unix

grgia committed:
Remove colors import from styles.js, replace with themeColors

vladamx committed:
Add html-inline-script-webpack-plugin

luacmartins committed:
fix report test

luacmartins committed:
fix unread tests

luacmartins committed:
fix ReportTest

luacmartins committed:
update dateutils tests

luacmartins committed:
fix test

grgia committed:
WIP clean up color imports for gray1-gray4 and appBG.

grgia committed:
Clean up themeColors references, particularly success/danger colors.

grgia committed:
Use dark theme for Onfido Component

grgia committed:
Clean up default.js and colors.js files

grgia committed:
Fix incorrect themeColors key in styles.js

Puneet-here committed:
use new key

Puneet-here committed:
clean up

vladamx committed:
Update to new-expensify.svg

Puneet-here committed:
remove unnecessary code

Co-authored-by: Sahil <thesahindia@gmail.com>

tgolen committed:
Refactor a few names to be more clear

tgolen committed:
Add code to group and sort the report options

tgolen committed:
Flatten all grouped reportIDs into a single array

tgolen committed:
Improve a few comments

tgolen committed:
Improve a few comments

tgolen committed:
Remove unused variables

dbond09 committed:
fix scroll speed of InvertedFlatList

luacmartins committed:
rename getLocalMomentFromDatetime

luacmartins committed:
rename datetimeToCalendarTime

luacmartins committed:
rename datetimeToRelative

luacmartins committed:
convert to utc time

tgolen committed:
Remove groupBy and only use arrays

luacmartins committed:
add more tests

luacmartins committed:
use created in unread tests

tgolen committed:
Ignore case when sorting by report name

tgolen committed:
Fix a bug with iou reports not sorting properly

youssef-lr committed:
Apply requested changes

youssef-lr committed:
Only update IOU if cancelled request has same currency as the report

tgolen committed:
Remove extra debugging

Puneet-here committed:
fix typo

Puneet-here committed:
add isDangerousAction prop

tgolen committed:
Protect against an empty IOU report collection

grgia committed:
Update Skeleton UI to dark theme colors

grgia committed:
Add blueLink and bring back branded blue in colors.js

sobitneupane committed:
Align emoji center in IOS

sobitneupane committed:
Fix lint error

grgia committed:
Add warning yellow to themeColors

tienifr committed:
fix: add m axSequenceNumber to optimistic

Puneet-here committed:
use ingresa instead of ingrese

Puneet-here committed:
add spanish copy

AndreasBBS committed:
Fix #12325

- In ReportActionItemMessageEdit.js change updateDraft's setState
- In ReportActionCompose.js change updateComment's setState
- In both changes, verify if the input has been manipulated.
If yes, mantain cursor position from previous state

Commit:
rebuild gh actions

vladamx committed:
Map to correct logo svg based on env

marcaaron committed:
Opt out of crashlytics with environment variable

marcaaron committed:
move into platform setup files

cristipaval committed:
Custom h1 tag style to make custom subject messages look nicer.

vladamx committed:
Update package.lock

vladamx committed:
Format webpack config and index.html

marcaaron committed:
remove the check in error boundary

VikalpP committed:
doc: add comments explaining the style changes

Puneet-here committed:
remove hyperlink functionality

aimane-chnaif committed:
prevent FAB menu and other page showing simultaneously

Beamanator committed:
Navigate back to profile after updating pronouns

Beamanator committed:
Update public pronouns translation

Beamanator committed:
Remove extra whitespace

Beamanator committed:
Show pronoun list directly on page

Beamanator committed:
Add ability to show option separator

roryabraham committed:
Make customairshipextender its own package

Beamanator committed:
Remove a few unneeded props

Beamanator committed:
Fix lint issues

roryabraham committed:
Dont use reportAction timestamp field in custom notification provider

Beamanator committed:
Add pronoun prefix to key

tienifr committed:
fix: issue

VikalpP committed:
doc: add reference link to UI changes

Co-authored-by: Rushat Gabhane <rushatgabhane@gmail.com>

fedirjh committed:
[Fix] Composer text and icon not aligned center vertically for native app

tienifr committed:
fix: comment

roryabraham committed:
Simplify catch

rushatgabhane committed:
fix: unclickable context menu

vladamx committed:
Overwrite svg size with higher selector specificity

OSBotify committed:
Update version to 1.2.27-1

OSBotify committed:
Update version to 1.2.27-2

Commit:
temporary fix for android zindex crash

Commit:
remove removed prop elsewhere

Commit:
address comments

yuwenmemon committed:
Remove renderCell method as it's no longer needed and causes errors

yuwenmemon committed:
Linter errors

Commit:
remove removeclippedsubviews altogether"

OSBotify committed:
Update version to 1.2.27-3

grgia committed:
Switch borderFocus to green, remove extra line

marcochavezf committed:
update structure Send money

marcochavezf committed:
update content

marcochavezf committed:
update content

Commit:
fix first field not focused after clicking "fix the errors" link on Form

AndreasBBS committed:
Fix formatting. npx eslint no errors

mountiny committed:
Bump expensify-common version

fedirjh committed:
Update styles.js

Add Comments to prevent regressions

parasharrajat committed:
Revert "workaround to hide blue border box on ReportActionItem"

marcaaron committed:
Remove waiting for initial app data

roryabraham committed:
Remove unnecessary BasePicker component

roryabraham committed:
Rename basePickerStyles to pickerStyles

roryabraham committed:
Suppress no-unused-prop-types for shouldSaveDraft

roryabraham committed:
Remove unnecessary clientID from ReportActionsList keyExtractor

roryabraham committed:
Prevent infinite loop in notificationPreference switcher

Beamanator committed:
Add a few additional pronouns

Beamanator committed:
Replace zeZieZirHir pronouns

Beamanator committed:
Fix pronoun list scrolling

Beamanator committed:
Bye bye annoying import

wildan-m committed:
Fix triple click behavior on last item chat

kadiealexander committed:
Add asterisk to support copy for #217678

Adding an asterisk for the spanish translation.

jayeshmangwani committed:
added isChronosReport prop to prevent edit and delete action for Chronos chat

OSBotify committed:
Update version to 1.2.27-4

Julesssss committed:
remove background colour applied to Android inputs on paste

Julesssss committed:
improve Android styles documentation

Pujan92 committed:
removed resetting the report fields to allow user to operate multiple actions on same popover

Puneet-here committed:
remove close account error modal and unused code

Puneet-here committed:
remove unused onyx prop

Puneet-here committed:
clear error on refresh and closing page

mountiny committed:
Update the comment

s77rt committed:
Fixed: Workspace - Error message for invalid phone number invite disappears fast

aimane-chnaif committed:
small refactor on SidebarLinks, BaseSidebarScreen

s77rt committed:
Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Jasper Huang <jasperhuangg@gmail.com>

s77rt committed:
Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Jasper Huang <jasperhuangg@gmail.com>

aimane-chnaif committed:
fix typo

aimane-chnaif committed:
fix typo

arosiclair committed:
Update README.md

parasharrajat committed:
Remove empty line

marcaaron committed:
Update CONTRIBUTING.md and PR Template to cover High Traffic and Offline situations

marcaaron committed:
lowercase

marcaaron committed:
undo whitespace changes

marcaaron committed:
remove more ws changes

s77rt committed:
Update WorkspaceMembersPage.js

OSBotify committed:
Update version to 1.2.28-0

s77rt committed:
Fixed: An emoji is entered multiple times on repeated presses without closing the emoji picker

aimane-chnaif committed:
add comments for didScreenBecomeInactive, isScreenInactive functions

marcaaron committed:
Fix contributing.md Slack link

wildan-m committed:
Add detailed comment for user-select-auto usage in appContainer

roryabraham committed:
Smarter regex for parsing No QA PRs

roryabraham committed:
Rebuild GH actions

Commit:
remove file

OSBotify committed:
Update version to 1.2.28-1

s77rt committed:
Update EmojiPicker.js

hellohublot committed:
fix camera image orientation

s77rt committed:
Update EmojiPicker.js

jayeshmangwani committed:
added isChronosReport & isArchivedRoom to defaultProps and propTypes

Commit:
add console logs to find where it is failing

s77rt committed:
Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Eugene <copyreading@gmail.com>

s77rt committed:
Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Eugene <copyreading@gmail.com>

s77rt committed:
Update WorkspaceMembersPage.js

vladamx committed:
Fix mapping of production logo  and do work after dom is ready

Puneet-here committed:
clear the error on refresh

OSBotify committed:
Update version to 1.2.28-2

jayeshmangwani committed:
added chronos condition in triggerHotkeyActions for edit action

hannojg committed:
chore: cleanup unused prop

aimane-chnaif committed:
add zeHirHirs pronoun

aldo-expensify committed:
Improve comment

Co-authored-by: Tim Golen <tgolen@gmail.com>

aldo-expensify committed:
Rename method / update documentation / refactor

Puneet-here committed:
add userEmailOrPhone inside validate function

grgia committed:
Make oldTheme / darkTheme accessible from  default.js

grgia committed:
Fix lint, update splash screen background color

grgia committed:
Adding cardBG key for other core branding PRs

Ollyws committed:
Added inline containerStyles to Tooltip

Ollyws committed:
Added WebkitUserSelect to userSelectText

AndrewGable committed:
Add iOS encrypted ad hoc provisioning profile

AndrewGable committed:
Fix typo

grgia committed:
Specify to use theme/default.js

OSBotify committed:
Update version to 1.2.29-0

OSBotify committed:
Update version to 1.2.29-1

OSBotify committed:
Update version to 1.2.29-2

OSBotify committed:
Update version to 1.2.29-3

Commit:
address comments

tgolen committed:
Make return types more consistent

tgolen committed:
Simplify checking for IOU reports

tgolen committed:
Prevent IOU reports from ever being null

Commit:
pass down error

grgia committed:
Fix button text

AndrewGable committed:
Add Browserstack API call to staging deploys

grgia committed:
Fix dropdown button line color

AndrewGable committed:
Mock firebase method

AndrewGable committed:
Move mock to correct file

grgia committed:
Fix default inverse color

OSBotify committed:
Update version to 1.2.29-4

marcaaron committed:
Add checklist item regarding changing function arguments

OSBotify committed:
Update version to 1.2.29-5

AndrewGable committed:
Attempt to fix iOS uploads to browserstack with different path

OSBotify committed:
Update version to 1.2.29-6

roryabraham committed:
Bump urbanairship-react-native version

roryabraham committed:
Install react-native-device-info package

roryabraham committed:
Add/update jest mocks

roryabraham committed:
Merge branch 'main' into Rory-TrackPushNotificationPermissions

# Conflicts:
#	.github/PULL_REQUEST_TEMPLATE.md
#	.github/actions/getReleaseBody/index.js
#	.github/actions/javascript/awaitStagingDeploys/index.js
#	.github/actions/javascript/checkDeployBlockers/index.js
#	.github/actions/javascript/getDeployPullRequestList/index.js
#	.github/actions/javascript/getPullRequestDetails/index.js
#	.github/actions/javascript/getReleaseBody/index.js
#	.github/actions/javascript/isPullRequestMergeable/index.js
#	.github/actions/javascript/isStagingDeployLocked/index.js
#	.github/actions/javascript/markPullRequestsAsDeployed/index.js
#	.github/actions/javascript/reopenIssueWithComment/index.js
#	.github/actions/javascript/triggerWorkflowAndWait/index.js
#	.github/actions/javascript/verifySignedCommits/index.js
#	.github/actions/markPullRequestsAsDeployed/index.js
#	.gitignore
#	README.md
#	__mocks__/urbanairship-react-native.js
#	contributingGuides/OFFLINE_UX.md
#	docs/_data/routes.yml
#	docs/_layouts/default.html
#	docs/index.html
#	ios/NewExpensify.xcodeproj/project.pbxproj
#	ios/Podfile.lock
#	jest/setup.js
#	package-lock.json
#	package.json
#	src/CONST.js
#	src/ONYXKEYS.js
#	src/ROUTES.js
#	src/components/AnchorForCommentsOnly/anchorForCommentsOnlyPropTypes.js
#	src/components/AnchorForCommentsOnly/index.js
#	src/components/ArrowKeyFocusManager.js
#	src/components/AttachmentModal.js
#	src/components/AvatarWithImagePicker.js
#	src/components/AvatarWithIndicator.js
#	src/components/Banner.js
#	src/components/Button.js
#	src/components/Composer/index.js
#	src/components/EmojiPicker/EmojiSkinToneList.js
#	src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js
#	src/components/HTMLEngineProvider/HTMLRenderers/PreRenderer/index.js
#	src/components/IOUConfirmationList.js
#	src/components/Icon/Expensicons.js
#	src/components/MenuItem.js
#	src/components/OfflineIndicator.js
#	src/components/OfflineWithFeedback.js
#	src/components/OnyxProvider.js
#	src/components/OptionsSelector/BaseOptionsSelector.js
#	src/components/OptionsSelector/index.android.js
#	src/components/OptionsSelector/index.js
#	src/components/OptionsSelector/optionsSelectorPropTypes.js
#	src/components/PDFView/index.native.js
#	src/components/ReportActionItem/IOUAction.js
#	src/components/ReportActionItem/IOUPreview.js
#	src/components/ReportWelcomeText.js
#	src/components/ScreenWrapper/index.android.js
#	src/components/ScreenWrapper/index.js
#	src/components/ScreenWrapper/propTypes.js
#	src/components/SettlementButton.js
#	src/components/TextInput/BaseTextInput.js
#	src/components/withWindowDimensions.js
#	src/languages/en.js
#	src/languages/es.js
#	src/libs/API.js
#	src/libs/Environment/betaChecker/index.android.js
#	src/libs/Middleware/Logging.js
#	src/libs/Middleware/SaveResponseInOnyx.js
#	src/libs/Navigation/AppNavigator/AuthScreens.js
#	src/libs/Navigation/AppNavigator/MainDrawerNavigator.js
#	src/libs/Navigation/AppNavigator/ModalStackNavigators.js
#	src/libs/Navigation/Navigation.js
#	src/libs/Navigation/NavigationRoot.js
#	src/libs/Navigation/linkingConfig.js
#	src/libs/NumberUtils.js
#	src/libs/OptionsListUtils.js
#	src/libs/Performance.js
#	src/libs/ReportUtils.js
#	src/libs/SidebarUtils.js
#	src/libs/actions/App.js
#	src/libs/actions/BankAccounts.js
#	src/libs/actions/FormActions.js
#	src/libs/actions/IOU.js
#	src/libs/actions/PaymentMethods.js
#	src/libs/actions/PersonalDetails.js
#	src/libs/actions/Policy.js
#	src/libs/actions/ReimbursementAccount/validateBankAccount.js
#	src/libs/actions/Report.js
#	src/libs/actions/ReportActions.js
#	src/libs/actions/Session/index.js
#	src/libs/actions/User.js
#	src/libs/actions/Wallet.js
#	src/libs/actions/Welcome.js
#	src/libs/deprecatedAPI.js
#	src/pages/AddPersonalBankAccountPage.js
#	src/pages/DetailsPage.js
#	src/pages/EnablePayments/ActivateStep.js
#	src/pages/EnablePayments/AdditionalDetailsStep.js
#	src/pages/EnablePayments/EnablePaymentsPage.js
#	src/pages/EnablePayments/TermsStep.js
#	src/pages/EnablePayments/userWalletPropTypes.js
#	src/pages/NewChatPage.js
#	src/pages/ReimbursementAccount/BankAccountManualStep.js
#	src/pages/ReimbursementAccount/BankAccountStep.js
#	src/pages/ReimbursementAccount/ReimbursementAccountForm.js
#	src/pages/ReimbursementAccount/ReimbursementAccountPage.js
#	src/pages/ReimbursementAccount/RequestorOnfidoStep.js
#	src/pages/ReimbursementAccount/RequestorStep.js
#	src/pages/ReimbursementAccount/ValidationStep.js
#	src/pages/RequestCallPage.js
#	src/pages/SetPasswordPage.js
#	src/pages/home/ReportScreen.js
#	src/pages/home/report/ContextMenu/ContextMenuActions.js
#	src/pages/home/report/ReportActionCompose.js
#	src/pages/home/report/ReportActionItemCreated.js
#	src/pages/home/report/ReportActionItemFragment.js
#	src/pages/home/report/ReportActionsList.js
#	src/pages/home/report/ReportActionsView.js
#	src/pages/home/report/ReportFooter.js
#	src/pages/home/sidebar/SidebarLinks.js
#	src/pages/iou/IOUCurrencySelection.js
#	src/pages/iou/IOUModal.js
#	src/pages/iou/IOUTransactions.js
#	src/pages/iou/steps/IOUConfirmPage.js
#	src/pages/settings/InitialSettingsPage.js
#	src/pages/settings/PasswordPage.js
#	src/pages/settings/Payments/AddDebitCardPage.js
#	src/pages/settings/Payments/PaymentMethodList.js
#	src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js
#	src/pages/settings/Payments/PaymentsPage/paymentsPagePropTypes.js
#	src/pages/settings/Payments/TransferBalancePage.js
#	src/pages/settings/PreferencesPage.js
#	src/pages/settings/Profile/ProfilePage.js
#	src/pages/signin/LoginForm.js
#	src/pages/signin/ResendValidationForm.js
#	src/pages/signin/SignInPageLayout/SignInPageContent.js
#	src/pages/workspace/WorkspaceBankAccountPage.js
#	src/pages/workspace/WorkspaceInitialPage.js
#	src/pages/workspace/WorkspaceInvitePage.js
#	src/pages/workspace/WorkspaceMembersPage.js
#	src/pages/workspace/WorkspaceNewRoomPage.js
#	src/pages/workspace/WorkspaceSettingsPage.js
#	src/pages/workspace/WorkspacesListPage.js
#	src/pages/workspace/reimburse/WorkspaceReimburseView.js
#	src/pages/workspace/withPolicy.js
#	src/styles/StyleUtils.js
#	src/styles/styles.js
#	tests/actions/ReportTest.js
#	tests/unit/NetworkTest.js
#	tests/unit/OptionsListUtilsTest.js
#	tests/unit/SidebarOrderTest.js
#	tests/utils/LHNTestUtils.js
#	tests/utils/TestHelper.js

roryabraham committed:
Reinstall packages after rebase + merge

roryabraham committed:
Fix diff after bad rebase + merge

roryabraham committed:
Remove accidental duplicate keys from package-lock.json

roryabraham commented:
Tried rebasing, now this PR says I want to merge 6500 commits 😮‍💨
